src/main/storm/mesos/MesosNimbus.java

```diff
@@ -114,94 +120,228 @@ public String getHostName(Map<String, SupervisorDetails> map, String nodeId) {
-    _localFileServerPort = Optional.fromNullable(port); 
+    _localFileServerPort = Optional.fromNullable(port);
   }
 
   protected void setupHttpServer() throws Exception {
-      _httpServer = new LocalFileServer();
-      _configUrl = _httpServer.serveDir("/conf", "conf", _localFileServerPort);
-      
-      LOG.info("Started HTTP server from which config for the MesosSupervisor's may be fetched. URL: " + _configUrl);
+    _httpServer = new LocalFileServer();
+    File dirPath = new File("generated-conf");
+    _configUrl = _httpServer.serveDir("/generated-conf", dirPath.getCanonicalPath(), _localFileServerPort);
+
+    LOG.info("Started HTTP server from which config for the MesosSupervisor's may be fetched. URL: " + _configUrl);
   }
 
   protected MesosSchedulerDriver createMesosDriver() throws IOException {
-      MesosSchedulerDriver driver;
-      Credential credential;
-      FrameworkInfo.Builder finfo = createFrameworkBuilder();
-      
-      if ((credential = getCredential(finfo)) != null) {
-          driver = new MesosSchedulerDriver(_scheduler,
-                                            finfo.build(),
-                                            (String) _conf.get(CONF_MASTER_URL),
-                                            credential);
-      } else {
-          driver = new MesosSchedulerDriver(_scheduler,
-                                            finfo.build(),
-                                            (String) _conf.get(CONF_MASTER_URL));
-      }     
-      
-      return driver;
+    MesosSchedulerDriver driver;
+    Credential credential;
+    FrameworkInfo.Builder finfo = createFrameworkBuilder();
+
+    if ((credential = getCredential(finfo)) != null) {
+      driver = new MesosSchedulerDriver(_scheduler,
+          finfo.build(),
+          (String) _conf.get(CONF_MASTER_URL),
+          credential);
+    } else {
+      driver = new MesosSchedulerDriver(_scheduler,
+          finfo.build(),
+          (String) _conf.get(CONF_MASTER_URL));
+    }
+
+    return driver;
   }
-  private OfferResources getResources(Offer offer, double executorCpu, double executorMem, double cpu, double mem) {
+
+  protected OfferResources getResources(Offer offer, double executorCpu, double executorMem, double cpu, double mem) {
     OfferResources resources = new OfferResources();
 
     double offerCpu = 0;
     double offerMem = 0;
 
     for (Resource r : offer.getResourcesList()) {
-      if (r.getName().equals("cpus") && r.getScalar().getValue() > offerCpu) {
-        offerCpu = r.getScalar().getValue();
-        resources.cpuSlots = (int) Math.floor((offerCpu - executorCpu) / cpu);
-      } else if (r.getName().equals("mem") && r.getScalar().getValue() > offerMem) {
-        offerMem = r.getScalar().getValue();
-        resources.memSlots = (int) Math.floor((offerMem - executorMem) / mem);
+      if (r.hasReservation()) {
+        // skip resources with reservations
+        continue;
```

Comment:
With the way dynamic reservations are implemented in Mesos, you may receive reserved offers for other frameworks. Since the Storm framework doesn't implement dynamic reservations, we just decline all of them.